build

```diff
@@ -942,7 +933,7 @@ def execute (message, cmd, working_dir=None):
 
   try:
     start = timer()
-    process = subprocess.Popen (cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=environ, cwd=working_dir)
+    process = subprocess.Popen (cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=environ, cwd=working_dir) #pylint: disable=consider-using-with
```

Comment:
Performing this operation with context management in Python2 resulted in compilation error messages not being written to the `error_text` variable, and hence no compilation debugging information printed. However in Python3 it worked for me. So should consider modifying the code here for `3.1.0` where Python2 support is dropped.